src/evaluate_location.py

```diff
@@ -29,27 +29,47 @@ def evaluate_location(location_data: dict) -> dict:
         location_data
     )
 
-    loc_feas = loc_res[loc_res['Feasible'] == 1].copy()
+    loc_feas = []
+    for key, value in loc_res.items():
+        if value['Feasible'] == 1:
+            loc_feas.append(value)
 
-    if not loc_feas.empty:
+    if len(loc_feas) > 0:
+        machines_value = []
+        for result in loc_feas:
+            machines_value.append(result['Machines'])
+        
         # find fewest feasible machines
-        mach_min = loc_feas['Machines'].min()
+        mach_min = min(machines_value)
 
         # keep the feasible setup with the fewest number of machines
-        loc_feas_min = loc_feas[loc_feas['Machines'] == mach_min].copy()
+        loc_feas_min = {}
+        for key, value in loc_res.items():
+            if value['Machines'] == mach_min:
+                loc_feas_min = value
+                break
```

Comment:
No need to loop twice here, just loop once to find the min machine num that is feasible and return it.